src/v8/v8_context_scope.rs

```diff
@@ -61,17 +65,83 @@ impl<'context_scope, 'data, 'isolate_scope, 'isolate, T: 'data> Drop
     }
 }
 
+/// A context scope as in the "Execution Context" scope. The main
+/// difference from the [V8IsolateScope] is that an `Isolate` is a
+/// lifetime boundary for some V8 Engine state we want to hold, and the
+/// [V8ContextScope] is the execution state of this state. An "isolate"
+/// is more like a storage of data and the code, and the
+/// [V8ContextScope] is the execution state for these data and code.
+///
+/// The [V8ContextScope] is more temporary, and so has more narrow
+/// lifetime, than an isolate (and so [V8IsolateScope]).
+///
+/// All the "local" values, such as, for example, [V8LocalString], are
+/// local to the isolate they were created in, and so to the
+/// [V8IsolateScope]. Such objects can't outlive the parent (their)
+/// isolate and so are always tied to the lifetime of those. To untie
+/// the connection between an object and its isolate, and to make such
+/// an object persistent and not temporary, as in "not tied to the
+/// parent isolate lifetime", it is possible to convert those to some
+/// abstract and serialised persisted values, such as
+/// [crate::v8::v8_value::V8PersistValue].
+#[derive(Debug)]
 pub struct V8ContextScope<'isolate_scope, 'isolate> {
-    pub(crate) inner_ctx_ref: *mut v8_context_ref,
-    pub(crate) exit_on_drop: bool,
-    pub(crate) isolate_scope: &'isolate_scope V8IsolateScope<'isolate>,
+    inner_ctx_ref: *mut v8_context_ref,
+    exit_on_drop: bool,
+    isolate_scope: &'isolate_scope V8IsolateScope<'isolate>,
 }
 
 impl<'isolate_scope, 'isolate> V8ContextScope<'isolate_scope, 'isolate> {
+    pub fn is_exit_on_drop(&self) -> bool {
```

Comment:
Why do we need to expose those? Maybe you just added them like a getter setter approach? If so can we make them `pub(crate)`? User should not touch them from outside.